content/sdk/10.js/00.ethers/20.v6/11.migration.md

```diff
@@ -1,32 +1,138 @@
+  - `Token.Address` has been removed.
+  - `Provider.getMessageProof` has been removed.
+  - `Provider.getTokenPrice` has been removed.
+
+- **Changes in Methods and Parameters:**
+  - `Provider.getBlockWithTransaction` has been replaced with `Provider.getBlock(<BlockTag>, true)`.
+  - `BlockWithTransaction` has been removed.
+
+- **Changes in Transaction Parameters:**
+  - `TransactionRequest.calldata` has been changed from `BytesLike` to `string`.
+  - `transaction.calldata` parameter in the following methods has been changed from `BytesLike` to `string`:
+    - `Provider.estimateL1ToL2Execute`
+    - `AdapterL1.getRequestExecuteTx`
+    - `AdapterL1.estimateGasRequestExecute`
+    - `AdapterL1.RequestExecute`
+
+- **Utilities:**
+  - `utils.parseTransaction` has been replaced by `utils.parseEip712`.
+
+### Migration Steps
+
+1. **Update Imports:**
+   Ensure you're importing from `zksync-ethers` instead of `zksync-web3`.
+
+   ```ts
+   import { Provider, Wallet, utils } from "zksync-ethers";
+   ```
+
+2. **Adapt to Method Changes:**
+   Replace deprecated methods with their new counterparts.
+
+   ```ts
+   // Old
+   const blockWithTx = await provider.getBlockWithTransaction(blockNumber);
+
+   // New
+   const blockWithTx = await provider.getBlock(blockNumber, true);
+   ```
+
+3. **Update Transaction Parameters:**
+   Change `calldata` parameters from `BytesLike` to `string`.
+
+   ```ts
+   // Old
+   const txRequest = {
+     calldata: ethers.utils.hexlify(data),
+   };
+
+   // New
+   const txRequest = {
+     calldata: data,
+   };
+   ```
+
+4. **Replace Utility Functions:**
+   Use the new utility functions provided by `ethers.js v6`.
+
+   ```ts
+   // Old
+   const parsedTx = utils.parseTransaction(rawTransaction);
+
+   // New
+   const parsedTx = utils.parseEip712(rawTransaction);
+   ```
+
+5. **Remove Deprecated Properties and Methods:**
+   Ensure you remove any usage of deprecated properties like `Token.Address` and methods such
+   as `Provider.getMessageProof` and `Provider.getTokenPrice`.
+
+### Example Migration
+
+#### Before Migration (Using `zksync-web3`)
+
+```ts
+import { Provider, Wallet, utils } from "zksync-web3";
+
+const provider = new Provider("https://mainnet.zksync.io");
+const wallet = new Wallet("<PRIVATE_KEY>", provider);
+
+const tx = {
+  to: "0xReceiverAddress",
+  value: ethers.utils.parseEther("1.0"),
+  calldata: ethers.utils.hexlify("0x1234"),
+};
+
+const gasEstimate = await provider.estimateL1ToL2Execute(tx);
+console.log(gasEstimate);
+
+const parsedTx = utils.parseTransaction("0xRawTransaction");
+console.log(parsedTx);
+```
+
+#### After Migration (Using `zksync-ethers v6`)
+
+```ts
+import { Provider, Wallet, utils } from "zksync-ethers";
+
+const provider = new Provider("https://mainnet.zksync.io");
+const wallet = new Wallet("<PRIVATE_KEY>", provider);
+
+const tx = {
+  to: "0xReceiverAddress",
+  value: ethers.utils.parseEther("1.0"),
```

Comment:
According to the ethers migration guide, it does not look like `ethers.utils` exists in V6.